connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectServer.scala

```diff
@@ -49,6 +49,7 @@ object SparkConnectServer extends Logging {
       SparkConnectService.server.awaitTermination()
     } finally {
       session.stop()
+      SparkConnectService.uiTab.foreach(_.detach())
```

Comment:
wouldn't SparkConnectService.stop() have been called already somewhere and already detached it?

Reply:
Does it get called on this path? I only saw `SparkConnectService.server.awaitTermination()` on line 49. When I searched for SparkConnectService.stop(), I only saw two occurrences in [SimpleSparkConnectService](https://github.com/apache/spark/blob/master/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/SimpleSparkConnectService.scala#L55) and [SparkConnectPlugin](https://github.com/apache/spark/blob/master/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/SparkConnectPlugin.scala)

Reply:
yeah... it seems that in this path the Server is killed by a signal (via `sbin/stop-connect-server.sh`) and then `SparkConnectService.stop()` is never called.
It would be nice to SparkConnectService.stop() here, but then it will try to shutdown the server again... 
In any case, it's unrelated to this PR, so let's leave it like it is now.